### DISCUSSION TITLE
Adding Scores to langfuse with dify

### DISCUSSION THREAD

**Discussion by shaquelsamir** [langfuse/langfuse#2640] [Support] [answered]
Hello All, I hope you are doing well.

I want a little help, I am new to dify and using self hosted dify and self hosted langfuse for monitoring.

I want to add scores like users' feedback - I know it is in dify but I want all the monitoring to be in langfuse, and other metrics using LLM as a judge but after adding the scores I didn't know how to link between them and dify, is there a way without editing dify or langfuse's code since I want to be always up-to-date with the latest version.

Thank you in advance

**Comment by marcklingen** [comment, marked as answer]
The dify `message_id` attribute maps to the langfuse `trace.id`. See docs here: https://langfuse.com/docs/integrations/dify#mapping-of-dify-to-langfuse

After using the dify api to create a new message, you can then add a score to the corresponding trace in Langfuse by attaching it to the trace with this id as the traceId, see docs here: https://langfuse.com/docs/scores/custom

Let me know in case you get stuck on the way, this should be fairly easy and I am happy to help

**Reply by shaquelsamir** [reply]
Thank you, but like this I need to add every new trace manually right?
in case I want the message to be evaluated automatically
also adding score in the configuration requires manual evaluation I can't use LLM as a judge
I saw the documents and I want something similar the beta version of the evaluation. making a pipeline won't work without editing dify's code.
If there is something not clear please let me know to explain my POV more.
Thank you

**Reply by marcklingen** [reply]
I understood now, you can build a pipeline as outlined here using the Langfuse API in case you do not want to use the langfuse-cloud beta: https://langfuse.com/docs/scores/model-based-evals#evaluation-pipeline

This works irrespective of the integration that you use as you rely on this api endpoint with time-based cursors: https://api.reference.langfuse.com/#get-/api/public/traces

**Reply by shaquelsamir** [reply]
Yes I saw it thank you
you mean I build a pipeline that every time extracts the traces and evaluates them and add the values to langfuse right?
create a scheduled operation that runs for example every hour and add the new score?
this is the only possible solution now if I don't want to use langfuse-cloud beta?

**Reply by marcklingen** [reply]
Yes, and we will add more example code on how to do this very soon as it is popular workflow to run custom evals on data in langfuse

**Reply by shaquelsamir** [reply]
Thank you a lot, one more thing for the user feedback, also we need to extract it. from dify and add it to langfuse or there is an automatic way ?

**Reply by marcklingen** [reply]
do you capture user feedback via dify? If so, you need to fetch it from their API

**Reply by Sos-Zachary** [reply]
> do you capture user feedback via dify? If so, you need to fetch it from their API您是否通过 Dify 收集用户反馈？如果是，您需要从他们的 API 中获取这些反馈。

https://langfuse.com/docs/sdk/typescript/guide-web

hi there! about the user feedback, Can i use langfuseWeb api to get it?is it a recommended way?
